src/maliput_osm/osm/connection.h

```diff
@@ -0,0 +1,57 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2022, Woven Planet.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <maliput/api/lane_data.h>
+
+#include "maliput_osm/osm/lane.h"
+
+namespace maliput_osm {
+namespace osm {
+
+/// Defines a connection between two lanes.
+/// The lanes can be connected from either end.
+/// Two connections are considered equal if they connect the same lanes using same ends without regard to the order.
+struct Connection {
+  /// The lane that is connected from.
+  osm::Lane::Id from_id;
+  /// The end of the lane that is connected from.
+  maliput::api::LaneEnd::Which from_end;
```

Comment:
Does it make sense to create a struct for an OSM "lane end"?

Reply:
We could do it for having a sort of analogy with maliput. However it won't be used by the users, it is just used for the builder. I don't have a strong opinion.
Do you want me to refactor this and create a LaneEnd struct? (Using lane::id instead of Lane* probably)

Reply:
It can be a nested struct too.
This is just a suggestion to provide more semantic encapsulation.

Reply:
check out [09294d6](https://github.com/maliput/maliput_osm/pull/33/commits/09294d6f5742c46fc26f8a55c1ea7d5fcdae3d93)